sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallingServerAsyncClient.java

```diff
@@ -0,0 +1,418 @@
+            });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Join a Call
+     *
+     * @param serverCallId The server call id.
+     * @param source to Join Call.
+     * @param joinCallOptions join call options.
+     * @return response for a successful join request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CallConnectionAsync> join(String serverCallId,
+                                           CommunicationIdentifier source,
+                                           JoinCallOptions joinCallOptions) {
+        try {
+            Objects.requireNonNull(serverCallId, "'serverCallId' cannot be null.");
+            Objects.requireNonNull(source, "'source' cannot be null.");
+            Objects.requireNonNull(joinCallOptions, "'joinCallOptions' cannot be null.");
+            return this.serverCallInternal
+                .joinCallAsync(serverCallId, JoinCallConverter.convert(source, joinCallOptions))
+                .onErrorMap(CommunicationErrorException.class, CallingServerErrorConverter::translateException)
+                .flatMap(response -> Mono.just(new CallConnectionAsync(response.getCallConnectionId(), callConnectionInternal)));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Join a Call
+     *
+     * @param serverCallId The server call id.
+     * @param source to Join Call.
+     * @param joinCallOptions join call options.
+     * @return response for a successful join request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CallConnectionAsync>>joinWithResponse(String serverCallId,
+                                                                CommunicationIdentifier source,
+                                                                JoinCallOptions joinCallOptions) {
+        try {
+            Objects.requireNonNull(serverCallId, "'serverCallId' cannot be null.");
+            Objects.requireNonNull(source, "'source' cannot be null.");
+            Objects.requireNonNull(joinCallOptions, "'joinCallOptions' cannot be null.");
+            return this.serverCallInternal.
+                joinCallWithResponseAsync(serverCallId, JoinCallConverter.convert(source, joinCallOptions))
+                .onErrorMap(CommunicationErrorException.class, CallingServerErrorConverter::translateException)
+                .map(response -> new SimpleResponse<>(response,
+                    new CallConnectionAsync(response.getValue().getCallConnectionId(), callConnectionInternal)));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<CallConnection> joinInternal(String serverCallId,
+                                                    CommunicationIdentifier source,
```

Comment:
Formatting issue. Can we align the params?

Reply:
formatting issue fixed